esp-link/cgitelnet.c

```diff
@@ -0,0 +1,79 @@
+#include <esp8266.h>
+#include "cgi.h"
+#include "config.h"
+#include "serbridge.h"
+
+// Cgi to return choice of Telnet ports
+int ICACHE_FLASH_ATTR cgiTelnetGet(HttpdConnData *connData) {
+  if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted
+
+  char buff[1024];
+  int len;
+
+  len = os_sprintf(buff,
+      "{ \"telnet-port1\":%d, \"telnet-port2\":%d }",
+      flashConfig.telnet_port1, flashConfig.telnet_port2);
+
+  jsonHeader(connData, 200);
+  httpdSend(connData, buff, len);
+  return HTTPD_CGI_DONE;
+}
+
+// Cgi to change choice of Telnet ports
+int ICACHE_FLASH_ATTR cgiTelnetSet(HttpdConnData *connData) {
+  if (connData->conn==NULL) {
+    return HTTPD_CGI_DONE; // Connection aborted
+  }
+
+  int8_t ok = 0;
+  uint8_t port1, port2;
+  ok |= getUInt16Arg(connData, "port1", &port1);
+  ok |= getUInt16Arg(connData, "port2", &port2);
+  if (ok < 0) return HTTPD_CGI_DONE;
+
+  char *coll;
+  if (ok > 0) {
+    // check whether ports are different
+    if (port1 == port2) { coll = "Ports cannot be the same!"; goto collision; }
+  
+    // we're good, set flashconfig
+    flashConfig.telnet_port1 = port1;
+    flashConfig.telnet_port2 = port2;
+    os_printf("Ports changed: port1=%d port2=%d\n",
+	port1, port2);
+
+    // apply the changes
+    //serbridgeReinit();
+    serbridgeInit(flashConfig.telnet_port1, flashConfig.telnet_port2);
```

Comment:
Please apply change after saving otherwise you might apply the changes, fail to save and return an error, but then things are inconsistent.